extract/extract/extractors/amazon.py

```diff
@@ -0,0 +1,321 @@
+        ]
+        return image_urls
+
+    return _handle_parse(targets, parse_image_urls)
+
+
+def _get_sizes(soup: BeautifulSoup) -> Optional[str]:
+    """
+    Helper function that extracts the product's sizes.
+
+    Args:
+        soup (BeautifulSoup): Parsed HTML
+
+    Returns:
+        Optional[str]: `str` object containing all sizes. If nothing was found `None`.
+    """
+    targets = [
+        soup.find_all("span", {"class", "a-size-base swatch-title-text-display swatch-title-text"})[
+            1:
+        ],
+        soup.find_all("option", id=re.compile("size_name"))[1:],
+    ]
+
+    def parse_sizes(sizes: list[BeautifulSoup]) -> str:
+        sizes = [size.text.strip() for size in sizes if size.text.strip()]
+        return ", ".join(sizes)
+
+    return _handle_parse(targets, parse_sizes)
+
+
+def _get_price(parsed_page: ParsedPage) -> Optional[float]:
+    """
+    Helper function that extracts the product's price.
+
+    Args:
+        parsed_page (ParsedPage): Intermediate representation of `ScrapedPage` domain object
+
+    Returns:
+        Optional[float]: `float` object if a price is given, else `None`.
+    """
+    targets = [
+        parsed_page.scraped_page.meta_information["price"],
+    ]
+
+    def parse_price(price: str) -> float:
+        return float(price.replace(".", "").replace(",", "."))
+
+    return _handle_parse(targets, parse_price)
+
+
+def _get_brand(soup: BeautifulSoup) -> Optional[str]:
+    """
+    Helper function that extracts the product's brand.
+
+    Args:
+        soup (BeautifulSoup): Parsed HTML
+
+    Returns:
+        Optional[str]: `str` object if a brand was found, else `None`.
+    """
+    targets = [
+        soup.find("div", {"id": "bylineInfo_feature_div"}).a.text,
+    ]
+
+    def parse_brand(brand: str) -> Optional[str]:
+        if brand.startswith("Besuche den "):
+            return brand[len("Besuche den ") : -len("-Store")]  # noqa
+        if brand.startswith("Marke: "):
+            return brand[len("Marke: ") :]  # noqa
+        return None
+
+    return (
+        _handle_parse(targets, parse_brand)
+        or _find_from_details_section(soup, "Marke")
+        or _find_from_details_section(soup, "Hersteller")
+    )
+
+
+def _get_description(soup: BeautifulSoup) -> str:
+    """
+    Helper function that extracts the product's description.
+
+    Args:
+        soup (BeautifulSoup): Parsed HTML
+
+    Returns:
+        str: `str` object with the description. Empty if nothing was found.
+    """
+    targets = [
+        soup.find("div", {"id": "productDescription"}),
+        soup.find("div", {"id": "feature-bullets"}),
+    ]
+
+    def parse_description(description: BeautifulSoup) -> str:
+        desc_paragraph = getattr(description, "p", None)
+        desc_list = description.find_all("li")
+        if desc_paragraph:
+            return desc_paragraph.get_text().strip()
+        if desc_list:
+            if "Mehr anzeigen" in desc_list[-1].text.strip():
```

Comment:
Same as above. This is probably a link? Can we use HTML here?

Reply:
Removed in https://github.com/calgo-lab/green-db/pull/65/commits/4c73798ad6e46ec3f0396227520ffde7c2888d12